lcls_tools/common/image/processing.py

```diff
@@ -0,0 +1,56 @@
+import numpy as np
+from scipy.ndimage import gaussian_filter
+from pydantic import BaseModel, PositiveFloat, ConfigDict
+from lcls_tools.common.image.roi import ROI
+
+
+class ImageProcessor(BaseModel):
+    """
+    Image Processing class that allows for background subtraction and roi cropping
+    ------------------------
+    Arguments:
+    roi: ROI (roi object either Circular or Rectangular),
+    background_image: np.ndarray (optional image that will be used in
+        background subtraction if passed),
+    threshold: Positive Float (value of pixel intensity to be subtracted
+        if background_image is None, default value = 0.0)
+    visualize: bool (plots processed image)
+    ------------------------
+    Methods:
+    subtract_background: takes a raw image and does pixel intensity subtraction
+    process: takes raw image and calls subtract_background, passes to result
+        to the roi object for cropping.
+    """
+
+    model_config = ConfigDict(arbitrary_types_allowed=True)
+    roi: ROI = None
+    background_image: np.ndarray = None
+    threshold: PositiveFloat = 0.0
+
+    def subtract_background(self, raw_image: np.ndarray) -> np.ndarray:
+        """Subtract background pixel intensity from a raw image"""
+        if self.background_image is not None:
+            image = raw_image - self.background_image
+        else:
+            image = raw_image - self.threshold
+        return image
+
+    def clip_image(self, image):
+        return np.clip(image, 0, None)
+
+    def process(self, raw_image: np.ndarray) -> np.ndarray:
+        """Process image by subtracting background pixel intensity
```

Comment:
I wish this function name were a little more informative, but maybe it's fine for now.

Reply:
I agree. Maybe "auto_process" gets the point across better?